### ISSUE TITLE
Tracking with learning: var referenced before assignment

### ISSUE THREAD

**Issue by wolny** [ilastik/ilastik#1756]
Happens, while computing the weights for the events.
Logs:
```
  File "/opt/ilastik-1.3.1b1-Linux/ilastik-meta/ilastik/ilastik/applets/tracking/structured/opStructuredTracking.py", line 302, in _runStructuredLearning
    if type == None:
```
UI error:
![trackingproblem](https://user-images.githubusercontent.com/706781/40058714-50eb921a-5852-11e8-8569-210a17858189.png)

**Comment by ilastikdev** [comment]
Hi,

Thanks for the suggestion,
but the new line type = None should be placed at line 248 (before the for loop).

Before we do that we should investigate why _type did not properly return a value or why the for loop was not entered, that is, why the trackSet is empty.

**Comment by wolny** [comment]
Good point, I don't have the original project file anymore, but will try to reproduce the issue. but yes, putting `type=None` before the `for` loop would not fix the root cause.
